components/webserver/utils/convert_html_to_header_file.sh

```diff
@@ -11,16 +11,15 @@ output_filename="page_${output_name}.h"
 echo """#ifndef PAGE_${output_name^^}_H
 #define PAGE_${output_name^^}_H
 
-const char page_${output_name,,}[] =""" > $output_filename
+// This file was generated using xxd""" > $output_filename
 
+# Gzip file and write to static array
+gzip --best "$1" -c > "page_${output_name}"
+xxd -i -u "page_${output_name}" >> $output_filename
+rm "page_${output_name}"
```

Comment:
can we pipe these commands like this?
 ```suggestion
gzip --best "$1" -c | xxd -i -u >> $output_filename
```

Reply:
Yea, that would be nice, making the extra file is a bit tacky. The problem is that `xxd` actually needs a complete file to get the length of the byte array and the variable names are derived from the filename too. If you manage to get it working, let me know and I am happy to change it.

Reply:
ok fair point. Sorry, I don't have build machine on hand to try it by myself atm, otherwise I would try that before asking you to change that 😁